infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -274,23 +275,24 @@ private void initializationTask() {
             }
 
             // Initiating periodic task to poll for failures.
-            try {
-                localMetricsPollingService.scheduleAtFixedRate(
-                        this::updateLocalMetrics,
-                        0,
-                        METRICS_POLL_INTERVAL.toMillis(),
-                        TimeUnit.MILLISECONDS);
-
-                detectionTasksScheduler.scheduleAtFixedRate(
-                        this::detectorTaskScheduler,
-                        0,
-                        policyExecuteInterval,
-                        TimeUnit.MILLISECONDS);
-            } catch (RejectedExecutionException err) {
-                log.error("Error scheduling failure detection task, {}", err);
-            }
+            localMetricsPollingService.scheduleAtFixedRate(
+                    this::updateLocalMetrics,
+                    0,
+                    METRICS_POLL_INTERVAL.toMillis(),
+                    TimeUnit.MILLISECONDS);
+
+            detectionTasksScheduler.scheduleAtFixedRate(
+                    this::detectorTaskScheduler,
+                    0,
+                    policyExecuteInterval,
+                    TimeUnit.MILLISECONDS);
+
         } catch (InterruptedException e) {
```

Comment:
Why do you differentiate between InterruptedException and Exception ? shouldn;t you throw an UnrecoverableCorfuError in both cases?